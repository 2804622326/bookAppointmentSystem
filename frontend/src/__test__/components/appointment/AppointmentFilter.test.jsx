import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AppointmentFilter from '../../../components/appointment/AppointmentFilter.jsx';

test('invokes callbacks and matches snapshot', () => {
  const onSelectStatus = jest.fn();
  const onClearFilters = jest.fn();
  const { asFragment } = render(
    <AppointmentFilter
      statuses={['pending', 'approved']}
      selectedStatus='pending'
      onSelectStatus={onSelectStatus}
      onClearFilters={onClearFilters}
    />
  );

  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'approved' } });
  expect(onSelectStatus).toHaveBeenCalledWith('approved');

  fireEvent.click(screen.getByRole('button', { name: /clear filter/i }));
  expect(onClearFilters).toHaveBeenCalled();

  expect(asFragment()).toMatchSnapshot();
});
