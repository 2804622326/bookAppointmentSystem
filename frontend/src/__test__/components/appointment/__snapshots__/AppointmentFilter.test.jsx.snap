// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invokes callbacks and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-5 container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 col-12"
      >
        <form
          class=""
        >
          <div>
            <label
              class="form-label"
            >
              Filter appointments by status:
            </label>
            <div
              class="input-group"
            >
              <select
                class="form-select"
              >
                <option
                  value="all"
                >
                  all
                </option>
                <option
                  value="pending"
                >
                  pending
                </option>
                <option
                  value="approved"
                >
                  approved
                </option>
              </select>
              <button
                class="btn btn-secondary"
                type="button"
              >
                Clear filter
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
